### PR TITLE
Add email type to email field in Checkout block

### LINKED ISSUE #48610: [Bug] The email field in the Checkout block has `type:text` instead of `type: email`
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Screenshot 2024-06-19 at 12 44 06
The email field in the Checkout block has `type:text`. 

This makes it harder for non-desktop users to fill in the information, as phone and tables optimize the keyboard depending on the input type. 

### Expected behavior

The fied should have `type:email` because:
- it's semantic 
- built in validation
- accesibility, better experience for screen readers
- browser features

### Actual behavior

The input is treated as a normal text field.

### Steps to reproduce

1. Go to the Checkout block
2. Inspect the Email address field under the Contact information section
3. Notice the type text
4. Go to a phone and notice that the email keyboard is not activated

### WordPress Environment

n/a

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by senadir**
Previous to additional fields, the email field was hardcoded in the contact information step, this also included its type, when we abstracted that, we started using fields from CheckoutFields, however those didn't include a type. This PR includes that.

Closes https://github.com/woocommerce/woocommerce/issues/48610

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. From a mobile phone (or via the link to the playground), add a product to cart and go to Checkout.
2. Focus on Email field, make sure an email keyboard is shown (`@` is a top level key in the keyboard).
3. If you have access to an ipad, make sure it also shows provider autocomplete buttons.
4. You can also inspect the page and see if the field type is email and autocomplete is email.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

> Mark Checkout email field as type email to show relevant keyboard when filling.

### CODE REVIEW

File: plugins/woocommerce/changelog/48611-add-type-to-email-field
Comment (mikejolley) [inline plugins/woocommerce/changelog/48611-add-type-to-email-field]: This needs clearing up.

**Review by mikejolley — APPROVED** [review]
I don't have access to an iPad but I did confirm it was correct in the source. Looks good 👍🏻 Just make sure you fix up the changelog before its merged because there is some HTML there that shouldn't be.

**Comment by jeffpaul** [comment]
@woocommerce/rubik looks like this PR should be good to merge/release, yeah?  If more is needed, let me know and we can try and help close the gap so this can get merged/released... thanks!

**Pull request merged**
